core/logger/splunk.go

```diff
@@ -0,0 +1,353 @@
+package logger
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+// Static audit log event type constants
+const (
+	AUTH_LOGIN_FAILED_EMAIL     = "AUTH_LOGIN_FAILED_EMAIL"
```

Comment:
These should probably be mixed caps to match standard Go naming https://go.dev/doc/effective_go#mixed-caps
```suggestion
	AuthLoginFailedEmail     = "AUTH_LOGIN_FAILED_EMAIL"
```
I was also wondering if it would be more clear to move them to a package in a sub dir, since they are specicially audit log event IDs, and not general log package constants.... :thinking: 

WDYT about using a custom type? e.g.
```go
type AuditEventID string

const (
    AuthLoginFailedEmail AuditEventID =  "AUTH_LOGIN_FAILED_EMAIL"
```
And then the API can be tightened up to accept `AuditEventID` instead of string.
```go
Audit(id AuditEventID, data map[string]interface{})
```

Reply:
Or, a combination of both could be a little simpler even:
```go
package audit

type EventID string

const (
    AuthLoginFailedEmail EventID = "AUTH_LOGIN_FAILED_EMAIL"
```
```go
Audit(id audit.EventID, data map[string]any)
```
```go
lggr.Audit(audit.AuthLoginFailedEmail, map[string]any{})
```

Reply:
I see what you mean as these constants pollute the logger namespace, so moved to logger/audit and changed constant name casing to upper camel case - added strong typing as well, good idea